### PR TITLE
feat(zk_toolbox): Run formatters and linterrs

### PR DESCRIPTION
**Pull request by Deniallugo**
## What ❔

Adding an ability to run linters and formatters for zk supervisor

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: zk_toolbox/crates/zk_supervisor/src/main.rs [hunk 1]
```diff
@@ -42,6 +42,8 @@ enum SupervisorSubcommands {
     Snapshot(SnapshotCommands),
     #[command(about = MSG_SUBCOMMAND_LINT_ABOUT, alias = "l")]
     Lint(LintArgs),
+    #[command(about = MSG_SUBCOMMAND_LINT_ABOUT)]
```
Comment (matias-gonz) [inline zk_toolbox/crates/zk_supervisor/src/main.rs:45]: This is using the same message as lint subcommand

File: zk_toolbox/crates/zk_supervisor/src/commands/fmt.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
+use std::path::PathBuf;
```
Comment (sanekmelnikov) [inline zk_toolbox/crates/zk_supervisor/src/commands/fmt.rs:1]: I store ecosystem folder inside zksync-era (e.g. `zksync-era/my-ecosystem`) and when I run `zk fmt` - everything works, but `zk_supervisor fmt` fails due to access to docker volumes
probably my use case isn't that common, but just letting you know

File: zk_toolbox/crates/zk_supervisor/src/messages.rs [hunk 3]
```diff
@@ -152,5 +156,18 @@ pub(super) fn msg_running_linter_for_extension_spinner(extension: &Extension) ->
     format!("Running linter for files with extension: .{}", extension)
 }
 
+pub(super) fn msg_running_fmt_for_extension_spinner(extension: Extension) -> String {
```
Comment (sanekmelnikov) [inline zk_toolbox/crates/zk_supervisor/src/messages.rs:159]: actually, i see this message twice (sometimes), is it ok?
Screenshot 2024-08-19 at 6 54 59 PM

File: zk_toolbox/crates/zk_supervisor/src/commands/fmt.rs (hunk 2 above)
Comment (Deniallugo) [reply zk_toolbox/crates/zk_supervisor/src/commands/fmt.rs:1]: ouch, that is the reason of old obscure code...

Comment (Deniallugo) [reply zk_toolbox/crates/zk_supervisor/src/commands/fmt.rs:1]: @sanekmelnikov could you try one more time please? 
seems i've fixed but it wasn't reproduced

**Review by matias-gonz — APPROVED** [review]

File: zk_toolbox/crates/zk_supervisor/src/commands/fmt.rs (hunk 2 above)
Comment (sanekmelnikov) [reply zk_toolbox/crates/zk_supervisor/src/commands/fmt.rs:1]: discussed offline, still not working - need to get rid of glob, mentioned [here](https://github.com/matter-labs/zksync-era/blob/main/etc/utils/src/index.ts#L128-L134)

**Review by sanekmelnikov — APPROVED** [review]

**Pull request merged**